code/game/machinery/_machinery.dm

```diff
@@ -179,6 +178,25 @@ Class Procs:
 /obj/machinery/proc/process_atmos()//If you dont use process why are you here
 	return PROCESS_KILL
 
+
+///Called when we want to change the value of the machine_stat variable. Holds bitflags.
+/obj/machinery/proc/set_machine_stat(new_value)
+	if(new_value == machine_stat)
+		return
+	. = machine_stat
+	machine_stat = new_value
+	on_set_machine_stat(.)
+
+
+///Called when the value of `machine_stat` changes, so we can react to it.
+/obj/machinery/proc/on_set_machine_stat(old_value)
+	if(old_value & (NOPOWER|BROKEN|MAINT))
+		if(!(machine_stat & (NOPOWER|BROKEN|MAINT))) //From off to on.
```

Comment:
What if make `(NOPOWER|BROKEN|MAINT)` a variable to be able make machinery that continue work when unpowered or on maint? With no need to override this proc. Or blocking of aquirring  NOPOWER flag by powerless machinery.

Reply:
A `machine_operational` variable would be good to have, along with a `set_machine_operational()` proc.
If you want to do that, it'd be an improvement.